src/services/recovery/__tests__/setup.test.ts

```diff
@@ -0,0 +1,100 @@
+import { getModuleInstance, KnownContracts, deployAndSetUpModule } from '@gnosis.pm/zodiac'
+import { faker } from '@faker-js/faker'
+import { BigNumber } from 'ethers'
+import type { Web3Provider } from '@ethersproject/providers'
+import type { SafeInfo } from '@safe-global/safe-gateway-typescript-sdk'
+
+import { getRecoverySetup } from '@/services/recovery/setup'
+
+jest.mock('@gnosis.pm/zodiac', () => ({
+  ...jest.requireActual('@gnosis.pm/zodiac'),
+  getModuleInstance: jest.fn(),
+  deployAndSetUpModule: jest.fn(),
+}))
+
+const mockGetModuleInstance = getModuleInstance as jest.MockedFunction<typeof getModuleInstance>
+const mockDeployAndSetUpModule = deployAndSetUpModule as jest.MockedFunction<typeof deployAndSetUpModule>
+
+describe('getRecoverySetup', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should deploy Delay Modifier, enable it on Safe and add a Guardian', () => {
```

Comment:
nit: I would suggest wording it a bit differently since the function call only returns those transactions and doesn't execute them

Also, is there a sad path that we should also test? e.g. if `deployAndSetUpModule` or `encodeFunctionData` throws?

Reply:
I've renamed the test in 0fb37869516077a163a979e0269b1eecacddd993. Regarding the sad path, I do not think it's necessary as it would be if x throws expect y to throw.